core/src/main/java/com/isbx/androidtools/media/ImageResizer.java

```diff
@@ -205,7 +224,22 @@ public Uri scaleImage(Uri sourceUri, ImageResizeConfig.Dimension targetDimension
                 os = context.openFileOutput(fileName, Context.MODE_PRIVATE);
 
                 ExifInterface exif = new ExifInterface();
-                exif.readExif(context.getContentResolver().openInputStream(sourceUri));
+
+                boolean isJpeg = false;
+                try {
+                    isJpeg = imageIsJPEG(sourceUri);
+                } catch(Exception e) {
+                    e.printStackTrace();
+                }
+                if (isJpeg) {
+                    exif.readExif(context.getContentResolver().openInputStream(sourceUri));
+                } else {
+                    //force bitmap as JPG before reading exif
+                    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                    out.compress(Bitmap.CompressFormat.JPEG, 100, stream);
+                    byte[] byteArray = stream.toByteArray();
+                    out = BitmapFactory.decodeByteArray(byteArray, 0, byteArray.length);
```

Comment:
@ebbnormal I don't think this step is necessary, as the exif writer will convert the bitmap to a jpeg automatically. Also we want to avoid loading the entire byte data of the image into memory at once